unit-tests-java8/src/test/java/org/eclipse/collections/test/map/UnmodifiableMapTestCase.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2021 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.test.map;
+
+import java.util.Map;
+
+import org.eclipse.collections.impl.tuple.ImmutableEntry;
+import org.eclipse.collections.test.map.mutable.MapTestCase;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.eclipse.collections.test.IterableTestCase.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+public interface UnmodifiableMapTestCase
+        extends MapTestCase
+{
+    @Test
+    default void Map_put()
+    {
+        Map<Integer, String> map = (Map<Integer, String>) this.newWithKeysValues(3, "Three", 2, "Two", 1, "One");
```

Comment:
newWithKeysValues() returns Object in these tests, to allow for Map and MapIterable to have the same assertions. The result of the method call is cast to Map.